iina/ViewLayer.swift

```diff
@@ -347,6 +347,40 @@ class ViewLayer: CAOpenGLLayer {
     return ctx
   }
 
+  // MARK: - ICC Profile
+
+  /// Set an ICC profile for use with the mpv [icc-profile-auto](https://mpv.io/manual/stable/#options-icc-profile-auto)
+  /// option.
+  ///
+  /// This method fulfills the mpv requirement that applications using libmpv with the render API provide the ICC profile via
+  /// `MPV_RENDER_PARAM_ICC_PROFILE` in order for the `--icc-profile-auto` option to work. The ICC profile data will not
+  /// be used by mpv unless the option is enabled.
+  ///
+  /// The IINA `Load ICC profile` setting is tied to the `--icc-profile-auto` option. This allows users to override IINA using
+  /// the [--icc-profile](https://mpv.io/manual/stable/#options-icc-profile) option.
+  func setRenderICCProfile(_ profile: NSColorSpace) {
+    videoView.$isUninited.withLock() { isUninited in
```

Comment:
I merged this PR into my own code yesterday and have since been getting frequent crashes at startup when using the Xcode debugger, coming from inside `mpv_render_context_render`. The last 2 I saved:
1. `Assertion failed: (tex->params.render_src), function update_uniform, file ra_gl.c, line 990.`
2. `Assertion failed: (h->canary == CANARY), function ta_dbg_check_header, file ta.c, line 291.`

I did some experiments and it looks like I fixed it by locking the OpenGL context around the `isUninited` lock. I realize my code is different than `develop` but I do suspect the problem exists here but just may be more rare. I suspect mpv needs calls to `mpv_render_context_set_parameter` to be isolated from `mpv_context_render` by the calling code.

Reply:
I will make this a draft and look into this.

Reply:
Yes, that was it. I added the code to lock the OpenGL context.